src/renderer/features/network/NetworkSelector/ui/NetworkSelector.tsx

```diff
@@ -1,93 +1,60 @@
 import { Listbox, Transition } from '@headlessui/react';
-import { useState, Fragment, useEffect } from 'react';
+import { Fragment } from 'react';
+import { TFunction } from 'react-i18next';
 
 import { cnTw } from '@shared/lib/utils';
 import { Icon, FootnoteText, IconButton, Button, HelpText } from '@shared/ui';
 import { useI18n } from '@app/providers';
-import { ExtendedChain } from '@entities/network';
+import type { ConnectionList, SelectorPayload } from '../lib/types';
 import { SelectButtonStyle, OptionStyle } from '@shared/ui/Dropdowns/common/constants';
 import { useScrollTo } from '@shared/lib/hooks';
 import { CommonInputStyles, CommonInputStylesTheme } from '@shared/ui/Inputs/common/styles';
 import { ConnectionType } from '@shared/core';
 import type { Theme } from '@shared/ui/types';
 import type { RpcNode } from '@shared/core';
 
-export const OptionsContainerStyle =
+const OptionsContainerStyle =
   'mt-1 absolute z-20 py-1 px-1 w-full border border-token-container-border rounded bg-input-background shadow-card-shadow';
 
 const TRANSITION_DURATION = 100;
 
-type SelectorPayload = {
-  type: ConnectionType;
-  title?: string;
-  node?: RpcNode;
+const Title = {
+  [ConnectionType.AUTO_BALANCE]: {
+    title: (t: TFunction) => t('settings.networks.selectorAutoBalance'),
+  },
+  [ConnectionType.DISABLED]: {
+    title: (t: TFunction) => t('settings.networks.selectorDisableNode'),
+  },
+  [ConnectionType.LIGHT_CLIENT]: {
+    title: (t: TFunction) => t('settings.networks.selectorLightClient'),
+  },
+  [ConnectionType.RPC_NODE]: {
+    title: (t: TFunction, nodeName?: string) => nodeName,
+  },
 };
 
 type Props = {
-  networkItem: ExtendedChain;
+  nodesList: ConnectionList[];
```

Comment:
`List` already claims to be an array, according to types it should result in array of arrays 😉
I think `connectionList: ConnectionItem[]` conveys the meaning better
And `selectedConnection?: ConnectionItem;` as well
Could you remind me why it's optional? If connection is disabled `selectedConnection` should have disabled option

Reply:
All in all these props are brilliant, exactly what I wanted to see in simple UI component

Reply:
It's because of the typeScript - we use `nodes.find` so one of the possible outcomes is undefined. To avoid it we can add additional checks like `|| {disabled selectedConnection}` or if we absolutely sure that it should be a return result from our `find` function - we can do like this 
`const selectedNode = nodes.find((node) => node.type === item.connection.connectionType)!`
adding a `!` in the end will solve the issue with typeScript

Reply:
Alright, let's leave it as it, in the end `find` returning `undefined` is a totally valid result